prober/http.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package prober
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+const maxHTTPBody = 4 << 20 // MiB
+
+// HTTP returns a Probe that healthchecks an HTTP URL.
+//
+// The Probe sends a GET request for url, expects an HTTP 200
+// response, and verifies that want is present in the response
+// body. If the URL is HTTPS, the probe further checks that the TLS
+// certificate is good for at least the next 7 days.
+func HTTP(url, wantText string) Probe {
```

Comment:
how about taking in a `func() *http.Request`? and add a helper for GET? would allow adding POSTs.


I don't know what we are probing, maybe not required.

Reply:
Hmm, possibly, but usage of these functions is already quite nested:

```
p.Run("my-probe", 10*time.Second, prober.HTTP("https://example.com", "Sign in"))
```

I'm worried about readability with a third level of nesting.

For right now, I'd like feature parity with our existing probers, which only do HTTP GET, TCP connect, and TLS validation. I'd say let's revisit when we need fancier HTTP stuff?